docs/strategy-customization.md

```diff
@@ -333,6 +333,22 @@ class AwesomeStrategy(IStrategy):
 
 ***
 
+#### Storing custom information using DatetimeIndex from `dataframe`
+
+Imagine you need to store an indicator like `ATR` or `RSI` into `custom_info`. To use this in a meaningful way, you will not only need the raw data of the indicator, but probably also need to keep the right timestamps.
+
+class AwesomeStrategy(IStrategy):
+    # Create custom dictionary
+    custom_info = {}
+
+    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
+        # add indicator mapped to correct DatetimeIndex to custom_info
+        # using "ATR" here as example
+        self.custom_info[metadata['pair']] = dataframe[['date', 'atr']].copy().set_index('date')
+        return dataframe
+
+See: (custom_stoploss example)[WIP] for how to access it
```

Comment:
we will want to fix that link for sure ... 


i also think we maybe should add a warning here - that this will increas memory consumption (especially relevant for backtesting / hyperopt).

Reply:
>i also think we maybe should add a warning here - that this will increas memory consumption (especially relevant for backtesting / hyperopt)

is it really that much more memory for a few more columns? maybe removing `.copy()` and just referencing it, would solve that.but it's been a while since I worked with python references and also I'm a new with dataframe, so I opted for the safer option by using `copy()` here. do you think we can remove it. How can I test it for edge cases.

Reply:
> we will want to fix that link for sure
done.

Reply:
well you're thinking about live - there, the memory consumption can be neglectible in most cases.

However in backtesting you might be talking about 2 columns ...
but what if that's 2 columns for 50 pairs for 500.000 lines (that's pretty easily doable on 1m timeframes).

and in hyperopt, you multiply the above result with the number of cores ... now it's no longer a "small amount of memory" - but can become a breaking factor for using hyperopt on a certain machine or not.

Reply:
@xmatthias do we really need to make a copy? Can not we keep a dataframe reference?

Reply:
well - it's just a documentation sample ... so it's up to the user how to do it in the end.

I'm ... not 100% sure on the impact - and in theory it can cause memory leaks - but if it really does (and under which circumstances) would need investigation. Unfortunately, it can also only appear after months of running (very slow memory leak) ... so it's rather dangerous to just tell users to store references to the dataframe "wherever" ... 

Honestly, while i think i understand memory management in python well enough - i do mistrust pandas in this regards, as it can EITHER return a copy or a view for simple operations (depending on some - to me - unclear rules) - so i think about every dataframe as a potential source of memory leaks.

Reply:
Given all those concerns, would it not make sense to provide a way to access internally stired dataframe instead of having user to store a reference or a copy or who knows what, leaving so much room for mistakes?

Reply:
during regular runs (dry / live) - the analyzed dataframe IS stored and available with `dp.get_analyzed_dataframe()`.

this is however problematic for backtesting / hyperopt where all "custom" columns are removed after populating buy / sell columns (for performance reasons) - so they would not be available, and 90% of the users will not need these.